rct_optimizations/test/covariance_utest.cpp

```diff
@@ -310,27 +368,31 @@ TEST_F(CircleFitUnit_TwoObsX, FitCircleToTwoPoints)
   printResults(result);
 
   // assert 3x3 covariance matrix
-  ASSERT_EQ(result.covariance.rows(), 3);
-  ASSERT_EQ(result.covariance.cols(), 3);
+  ASSERT_EQ(result.covariance.covariance_matrix.rows(), 3);
+  ASSERT_EQ(result.covariance.covariance_matrix.cols(), 3);
 
   // expect matrix to be symmetric
   // BUG: flaky way to evaluate, since difference is sometimes (but rarely) greater than epsilon
-  EXPECT_NEAR(result.covariance(0, 1), result.covariance(1, 0), std::numeric_limits<double>::epsilon());
-  EXPECT_NEAR(result.covariance(0, 2), result.covariance(2, 0), std::numeric_limits<double>::epsilon());
-  EXPECT_NEAR(result.covariance(1, 2), result.covariance(2, 1), std::numeric_limits<double>::epsilon());
+  EXPECT_NEAR(result.covariance.covariance_matrix(0, 1), result.covariance.covariance_matrix(1, 0), std::numeric_limits<double>::epsilon());
+  EXPECT_NEAR(result.covariance.covariance_matrix(0, 2), result.covariance.covariance_matrix(2, 0), std::numeric_limits<double>::epsilon());
+  EXPECT_NEAR(result.covariance.covariance_matrix(1, 2), result.covariance.covariance_matrix(2, 1), std::numeric_limits<double>::epsilon());
 
   // expect diagonal elements to be positive
-  EXPECT_GE(result.covariance(0, 0), 0.0);
-  EXPECT_GE(result.covariance(1, 1), 0.0);
-  EXPECT_GE(result.covariance(2, 2), 0.0);
+  EXPECT_GE(result.covariance.covariance_matrix(0, 0), 0.0);
+  EXPECT_GE(result.covariance.covariance_matrix(1, 1), 0.0);
+  EXPECT_GE(result.covariance.covariance_matrix(2, 2), 0.0);
 
-  // expect covariance between Y and R to be close to 1
-  EXPECT_NEAR(abs(result.covariance(1, 2)), 1.0, 1e-5);
+  // expect covariance.covariance_matrix between Y and R to be close to 1
+  EXPECT_NEAR(abs(result.covariance.correlation_matrix(1, 2)), 1.0, 1e-5);
```

Comment:
Use `fabs` or `std::abs` instead of `abs`. This expectation failed several times for me because it's comparing the values as integers